### DISCUSSION TITLE
Add 'global preferences' for an individual user

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5894] [Feature Requests]
>When editing the preferences in Sp7, the changes only applies to the current collection. Our institution have several collections, and as an admin who frequently changes collections, it is annoying to have to set for example font size repeatedly for all of them.

> To me, the nature of the preferences (font size, colours etc) are such that they feel intuitively global. It is still my point that they should be shared across collections.

**Discussed here:** https://discourse.specifysoftware.org/t/global-preferences/1444

**Requested By:** Chris at Gothenburg MNH

**Comment by specifysoftware** [comment]
This issue has been mentioned on **Specify Community Forum**. There might be relevant details there:

https://discourse.specifysoftware.org/t/global-preferences/1444/4

**Comment by maxpatiiuk** [comment]
You can create DefaultUserPreferences at a highest level, and they would apply to all levels below that
the xml-editor branch does a better job of introducing collection admins to all the possible app resources available for them, including DefaultUserPreferences (before this, you had to look at our wiki page to find out about this)

We went with "applying all user preferences at a collection level only" as that simplified things quite a bit:
- no need for us to decide which level something has to go on (i.e is accent color collection specific or not - some people like having different accent color in each collection)
- no need to communicate to the user what is scoped to collection, and what is global (sp6 prefs don't communicate these things at all so you have to guess)
- simpler in the code to have all user preferences in a single file, rather than fetch them at global level, and at collection level (and then merge if needed)
  - but now that user preferences are growing and growing in importance, it would be totally reasonable to invest more dev time into making sure this system better fulfills common use cases
  - for example, how about making user preferences apply globally by default, but then on-demand user could overwrite some preferences on a collection level only? (a simple UI for this is yet to be determined)
    - could use VS Code's settings for inspiration (they have Workspace and User levels)

**Comment by specifysoftware** [comment]
This issue has been mentioned on **Specify Community Forum**. There might be relevant details there:

https://discourse.specifysoftware.org/t/standardize-color-scheme-for-all-users-at-the-collection-level/1471/2
